tests/linearizability/model_test.go

```diff
@@ -402,6 +402,122 @@ func TestModelStep(t *testing.T) {
 				{req: txnRequest("key", "8", "10"), resp: txnResponse(false, 9)},
 			},
 		},
+		{
+			name: "Put with valid lease id should succeed. Put with invalid lease id should fail",
+			operations: []testOperation{
+				{req: leaseGrantRequest(1), resp: leaseGrantResponse(1)},
+				{req: putWithLeaseRequest("key", "2", 1), resp: putResponse(2)},
+				{req: putWithLeaseRequest("key", "3", 2), resp: putResponse(3), failure: true},
+				{req: getRequest("key"), resp: getResponse("2", 2)},
+			},
+		},
+		{
+			name: "Put with valid lease id should succeed. Put with expired lease id should fail",
+			operations: []testOperation{
+				{req: leaseGrantRequest(1), resp: leaseGrantResponse(1)},
+				{req: putWithLeaseRequest("key", "2", 1), resp: putResponse(2)},
+				{req: getRequest("key"), resp: getResponse("2", 2)},
+				{req: leaseRevokeRequest(1), resp: leaseRevokeResponse(3)},
+				{req: putWithLeaseRequest("key", "4", 1), resp: putResponse(4), failure: true},
+				{req: getRequest("key"), resp: getResponse("", 3)},
+			},
+		},
+		{
+			name: "Revoke should increment the revision",
+			operations: []testOperation{
+				{req: leaseGrantRequest(1), resp: leaseGrantResponse(1)},
+				{req: putWithLeaseRequest("key", "2", 1), resp: putResponse(2)},
+				{req: leaseRevokeRequest(1), resp: leaseRevokeResponse(3)},
+				{req: getRequest("key"), resp: getResponse("", 3)},
+			},
+		},
+		{
+			name: "Put following a PutWithLease will detach the key from the lease",
+			operations: []testOperation{
+				{req: leaseGrantRequest(1), resp: leaseGrantResponse(1)},
+				{req: putWithLeaseRequest("key", "2", 1), resp: putResponse(2)},
+				{req: putRequest("key", "3"), resp: putResponse(3)},
+				{req: leaseRevokeRequest(1), resp: leaseRevokeResponse(3)},
+				{req: getRequest("key"), resp: getResponse("3", 3)},
+			},
+		},
+		{
+			name: "Change lease. Revoking older lease should not increment revision",
+			operations: []testOperation{
+				{req: leaseGrantRequest(1), resp: leaseGrantResponse(1)},
+				{req: leaseGrantRequest(2), resp: leaseGrantResponse(1)},
+				{req: putWithLeaseRequest("key", "2", 1), resp: putResponse(2)},
+				{req: putWithLeaseRequest("key", "3", 2), resp: putResponse(3)},
+				{req: leaseRevokeRequest(1), resp: leaseRevokeResponse(3)},
+				{req: getRequest("key"), resp: getResponse("3", 3)},
+				{req: leaseRevokeRequest(2), resp: leaseRevokeResponse(4)},
+				{req: getRequest("key"), resp: getResponse("", 4)},
+			},
+		},
+		{
+			name: "Update key with same lease",
+			operations: []testOperation{
+				{req: leaseGrantRequest(1), resp: leaseGrantResponse(1)},
+				{req: putWithLeaseRequest("key", "2", 1), resp: putResponse(2)},
+				{req: putWithLeaseRequest("key", "3", 1), resp: putResponse(3)},
+				{req: getRequest("key"), resp: getResponse("3", 3)},
+			},
+		},
+		{
+			name: "Deleting a leased key - revoke should not increment revision",
```

Comment:
One additional scenario would be useful, "Overriding a leased key should remove lease"

Reply:
there is a scenario for Put following PutWithLease. https://github.com/etcd-io/etcd/blob/main/tests/linearizability/model_test.go#L435. Do you mean something else? Can you please clarify what you mean by override? thanks